src/vsr/grid_scrubber.zig

```diff
@@ -156,6 +159,71 @@ pub fn GridScrubberType(comptime Forest: type) type {
             scrubber.* = undefined;
         }
 
+        pub fn open(scrubber: *GridScrubber, random: std.rand.Random) void {
+            // Compute the tour origin exactly once.
+            if (scrubber.tour_tables_origin) |_| {
+                return;
+            }
+
+            // Each replica's scrub origin is chosen independently.
+            // This reduces the chance that the same block across multiple replicas can bitrot
+            // without being discovered and repaired by a scrubber.
+            //
+            // To accomplish this, try to select an origin uniformly across all blocks:
+            // - Bias towards levels with more tables.
+            // - Bias towards trees with larger Values (since their tables have more blocks).
+            // - (Though, for ease of implementation, the origin is always at the beginning of a
+            //   tree's level, never in the middle.)
+            assert(scrubber.tour == .init);
+
+            const tree_count = Forest.tree_infos.len;
+
+            var weights: [tree_count][constants.lsm_levels]u64 = undefined;
+            var weights_sum: u64 = 0;
+            for (0..constants.lsm_levels) |level| {
+                inline for (Forest.tree_infos, 0..) |tree_info, tree_index| {
+                    const tree_id = comptime Forest.tree_id_cast(tree_info.tree_id);
+                    const tree = scrubber.forest.tree_for_id_const(tree_id);
+                    const levels = &tree.manifest.levels;
+                    weights[tree_index][level] =
+                        levels[level].tables.len() * @sizeOf(tree_info.Tree.Table.Value);
```

Comment:
shouldn't be this additionally weighted by table_value_count_max?

Reply:
Id also maybe add an explicit `@as(u64)` here somewhere. I _think_ as stated the computation is carried over in `usize`, and we want u64 here.

Reply:
I'm going to combine `@sizeOf(Value)` and `table_value_count_max` into:

```
                    const tree_level_weight =
                        @as(u64, levels[level].tables.len()) *
                        tree_info.Tree.Table.index.data_block_count_max;
```

since that is more directly what we care about.